internal/metrics/cycletime/service.go

```diff
@@ -139,6 +139,10 @@ func (c *cycleTime) mergeRequestFirstCommit(commits []*models.Commit) *models.Co
 }
 
 func (c cycleTime) mergeRequestFirstComment(comments []*models.Comment) *models.Comment {
+	//TODO: not sure about that null check.
+	if len(comments) == 0 {
+		return &models.Comment{}
```

Comment:
Yes. Logic is contain null checks a one top layer. if the return is null, the calculate is cancelled, so null return is important.